recipes/bdwgc/all/conanfile.py

```diff
@@ -119,30 +129,42 @@ def _extract_copyright(self):
 
     def package(self):
         save(self, os.path.join(self.package_folder, "licenses", "COPYRIGHT"), self._extract_copyright())
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
         cmake.install()
         rmdir(self, os.path.join(self.package_folder, "share"))
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
-    @property
-    def _libs(self):
-        libs = []
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "BDWgc")
+        self.cpp_info.set_property("cmake_target_name", "BDWgc::BDWgc")
+
+        # TODO: Remove on Conan 2.0
+        self.cpp_info.names["cmake_find_package"] = "BDWgc"
+        self.cpp_info.names["cmake_find_package_multi"] = "BDWgc"
+
+        self.cpp_info.components["gc"].set_property("cmake_target_name", "BDWgc::gc")
+        self.cpp_info.components["gc"].set_property("pkg_config_name", "bdw-gc")
+        self.cpp_info.components["gc"].libs = ["gc"]
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.components["gc"].system_libs = ["pthread", "dl"]
+        self.cpp_info.components["gc"].defines = ["GC_DLL" if self.options.shared else "GC_NOT_DLL"]
+        if self.options.gc_debug:
+            self.cpp_info.components["gc"].defines.append("GC_DEBUG")
+        if self.settings.os == "Windows":
+            self.cpp_info.components["gc"].requires = ["libatomic_ops::libatomic_ops"]
+
         if self.options.cplusplus and self.options.get_safe("throw_bad_alloc_library"):
-            libs.append("gctba")
+            self.cpp_info.components["gctba"].set_property("cmake_target_name", "BDWgc::gctba")
```

Comment:
I think it is better to put libgctba.so to gccpp component (in addition to libgccpp.so). See my comment in https://github.com/andyli/libgc-cygwin/pull/9#issuecomment-1234162506 for details.
I will fix it if no objections.

Reply:
But it is unclear how to specify multiple values for cmake_target_name, I guess the following syntax is invalid:
self.cpp_info.components["gccpp"].set_property("cmake_target_name", ""BDWgc::gccpp;BDWgc::gctba")

If not possible to set multiple target for a component, let's stay with 2 components - gccpp & gctba

Reply:
> But it is unclear how to specify multiple values for cmake_target_name

It's impossible. Please, read the documentation to understand how properties work: https://docs.conan.io/en/latest/migrating_to_2.0/properties.html#properties-migration

Reply:
> I think it is better to put libgctba.so to gccpp component (in addition to libgccpp.so). 

The recipe follows exactly what is in the upstream's cmake. In case you want to use a different module, please, update your project and release a new version.

Reply:
Got it. Thank you for the explanation.